src/main/java/hudson/plugins/sonar/MsBuildSQRunnerInstallation.java

```diff
@@ -95,17 +99,42 @@ public String getDisplayName() {
     }
   }
 
-  public String getExecutable(Launcher launcher) throws IOException, InterruptedException {
+  public static String getScannerName() {
+    if (testExeName != null) {
+      return testExeName;
+    } else {
+      return SCANNER_EXE_NAME;
+    }
+  }
+
+  public String getToolPath(Launcher launcher) throws IOException, InterruptedException {
     return launcher.getChannel().call(new MasterToSlaveCallable<String, IOException>() {
       private static final long serialVersionUID = 1L;
 
       @Override
       public String call() throws IOException {
         String home = Util.replaceMacro(getHome(), EnvVars.masterEnvVars);
-        File exe = new File(home, exeName);
-        if (exe.exists()) {
-          return exe.getPath();
+
+        if (testExeName != null) {
```

Comment:
This is used only by tests hence the renaming. I put it first so that we can ensure tests can take precedence over whatever is in the path.

Besides I don't test that the tool exists so that we can test the behavior produced by a tool not found.